lib/osop/forecast.py

```diff
@@ -0,0 +1,213 @@
+import xarray as xr
+import xarray as xr
+import pandas as pd
+import numpy as np
+import eccodes
+import matplotlib.pyplot as plt
+
+
+
+
+### Import The files for practice: the data is using 
+# Month: 5 - May
+# Leads: 2,3,4  - June, July, August
+# Variable: t2m 
+# Services: 9
+
+#path to the .nc file
+MF_HC_TT = '/data/scratch/eleanor.dean/seafoam/data/master/hindcast/products/meteo_france_9_1993-2016_monthly_mean_5_234_45:-30:-2.5:60_2m_temperature.1m.tercile_thresholds.nc'
+MF_HC_ = '/data/scratch/eleanor.dean/seafoam/data/master/hindcast/downloads/meteo_france_9_1993-2016_monthly_mean_5_234_45:-30:-2.5:60_2m_temperature.grib'
+MF_FC_ = '/data/scratch/eleanor.dean/seafoam/data/master/forecast/downloads/meteo_france_9_2025-2025_monthly_mean_5_234_45:-30:-2.5:60_2m_temperature.grib'
+# Open the file as xr dataset
+MF_HC_TT = xr.open_dataset(MF_HC_TT)
+MF_HC = xr.open_dataset(MF_HC_)
+MF_FC = xr.open_dataset(MF_FC_)
+#Check the dataset info is correct
+print("this is hctt:",MF_HC_TT.t2m.size)
+print("this is hc:",MF_HC)
+print("this is fc:",MF_FC)
+
+
+### STOLEN FUNCTIONS from compute_products -- reformates the grib FC to be the same as the TC
```

Comment:
We seem to have 3 versions of some these functions now? please refactor so we only have one copy perhaps in lib/osop/util.py?